File: skills/render-create/templates/background-tasks/ts/cron.ts
/**
 * Cron Job
 *
 * Scheduled task that runs at specified intervals.
 * Executes once and exits.
 */

import "dotenv/config";

/**
 * Main cron job logic
 */
async function runCronJob(): Promise<void> {
  const startTime = Date.now();
  console.log(`Cron job started at ${new Date().toISOString()}`);

  try {
    // TODO: Add your scheduled task logic here
    // Examples:
    // - Clean up old records
    // - Send scheduled notifications
    // - Generate reports
    // - Sync data between systems

    console.log("Executing scheduled task...");

    // Placeholder: simulate work
    await new Promise((resolve) => setTimeout(resolve, 1000));

    console.log("Scheduled task completed successfully");
  } catch (error) {
    console.error("Cron job failed:", error);
    throw error;
  } finally {
    const duration = Date.now() - startTime;
    console.log(`Cron job finished in ${duration}ms`);
  }
}

// Run the cron job
runCronJob()
  .then(() => {
    console.log("Cron job exiting successfully");
    process.exit(0);
  })
  .catch((error) => {
    console.error("Cron job exiting with error:", error);
    process.exit(1);
  });

File: skills/render-create/templates/background-tasks/py/cron.py
"""
Cron Job

Scheduled task that runs at specified intervals.
Executes once and exits.
"""

import sys
import time
from datetime import datetime

from dotenv import load_dotenv

load_dotenv()


def run_cron_job() -> None:
    """Main cron job logic."""
    start_time = time.time()
    print(f"Cron job started at {datetime.now().isoformat()}")

    try:
        # TODO: Add your scheduled task logic here
        # Examples:
        # - Clean up old records
        # - Send scheduled notifications
        # - Generate reports
        # - Sync data between systems

        print("Executing scheduled task...")

        # Placeholder: simulate work
        time.sleep(1)

        print("Scheduled task completed successfully")

    except Exception as e:
        print(f"Cron job failed: {e}")
        raise

    finally:
        duration = time.time() - start_time
        print(f"Cron job finished in {duration:.2f}s")


if __name__ == "__main__":
    try:
        run_cron_job()
        print("Cron job exiting successfully")
        sys.exit(0)
    except Exception as e:
        print(f"Cron job exiting with error: {e}")
        sys.exit(1)
